pkg/accounts/kubernetes/parse.go

```diff
@@ -38,10 +40,49 @@ func (k *AccountType) FromSpinnakerConfig(settings map[string]interface{}) (acco
 	} else {
 		return nil, fmt.Errorf("name is not a string")
 	}
+	auth, err := k.authFromSpinnakerConfig(a.Name, settings)
+	if err != nil {
+		return nil, err
+	}
+	a.Auth = auth
 	a.Settings = settings
 	return a, nil
 }
 
+func (k *AccountType) authFromSpinnakerConfig(name string, settings map[string]interface{}) (*v1alpha2.KubernetesAuth, error) {
+	kubeconfigFile, ok := settings["kubeconfigFile"]
+	if ok {
+		s, sok := kubeconfigFile.(string)
+		if !sok {
+			return nil, fmt.Errorf("kubeconfigFile is not a string: %s", kubeconfigFile)
+		}
+		return &v1alpha2.KubernetesAuth{KubeconfigFile: s}, nil
+	}
+	sa, ok := settings["serviceAccount"]
+	if ok {
+		s, sok := sa.(bool)
+		if !sok {
+			return nil, fmt.Errorf("serviceAccount is not a boolean: %s", sa)
+		}
+		return &v1alpha2.KubernetesAuth{UseServiceAccount: s}, nil
+	}
+	kubeContent, ok := settings["kubeconfig"]
```

Comment:
What is `kubeconfig`? Is it for `kubeconfigFileContent`?

Reply:
Misnamed, should be `kubeconfigContents`